Sources/SotoCore/Credential/RotatingCredentialProvider.swift

```diff
@@ -24,17 +24,46 @@ import struct Foundation.TimeInterval
 ///
 /// Used for wrapping another credential provider whose `getCredential` method returns an `ExpiringCredential`.
 /// If no credential is available, or the current credentials are going to expire in the near future  the wrapped credential provider
-/// `getCredential` is called again. If current credentials have not expired they are returned otherwise we wait on new
+/// `getCredential` is called again. If current credentials have not expired (within a threshold) they are returned otherwise we wait on new
 /// credentials being provided.
 public final class RotatingCredentialProvider: CredentialProvider {
     let expiringCredential: ExpiringValue<Credential>
-
+    let validCredentialThreshold: TimeInterval
     public let provider: CredentialProvider
 
-    public init(context: CredentialProviderFactory.Context, provider: CredentialProvider, remainingTokenLifetimeForUse: TimeInterval? = nil) {
+    ///  Initialize RotatingCredentialProvider
+    /// - Parameters:
+    ///   - context: Context used to create this credential provider
+    ///   - provider: Credential provider to request credentials from
+    ///   - remainingTokenLifetimeForUse: How near to expiration, before we request new credentials
+    public init(
+        context: CredentialProviderFactory.Context,
+        provider: CredentialProvider,
+        remainingTokenLifetimeForUse: TimeInterval? = nil
+    ) {
+        self.provider = provider
+        self.validCredentialThreshold = 15
+        self.expiringCredential = .init(threshold: remainingTokenLifetimeForUse ?? 165) {
```

Comment:
Why 165 here when the below is using 180? Would it be easier to call the new initialiser with the `validCredentialThreshold`?

Reply:
Previously it was 180 seconds. Now it is 165 seconds + 15 seconds from the validCredentialThreshold

Reply:
Ah yeah makes sense 👍